fortls/ftypes.py

```diff
@@ -0,0 +1,123 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import NamedTuple
+
+#: A single line range tuple
+Range = NamedTuple("Range", [("start", int), ("end", int)])
```

Comment:
Maybe make it a general range which default to single line if nothing else is specified?